timelock-server/build.gradle

```diff
@@ -42,6 +42,13 @@ dependencies {
     integTestCompile group: 'io.dropwizard', name: 'dropwizard-testing'
 }
 
+task prepareForEteTests(type: Copy, dependsOn: 'distTar') {
```

Comment:
For the gradle integration, check out https://github.com/palantir/gradle-docker.

One thing it will do for you is to create a directory with only your latest .tgz, which will allow you to use `ADD timelock-server-*.tgz` and get rid of this extra task.

Later down the line, it will also avoid a lot of boilerplate for the publishing of the docker image.